plutus-core/plutus-ir/test/PlutusIR/Generators/QuickCheck/Tests.hs

```diff
@@ -210,7 +210,8 @@ noStructuralErrors term =
 prop_noStructuralErrors :: Property
 prop_noStructuralErrors = withMaxSuccess 99 $
   forAllDoc "ty,tm" genTypeAndTerm_ shrinkClosedTypedTerm $ \(_, termPir) -> ioProperty $ do
-    termUPlc <- fmap UPLC._progTerm . modifyError throw . toUPlc $ Program () latestVersion termPir
+    termUPlc <- fmap UPLC._progTerm . modifyError (userError . displayException) . toUPlc $
```

Comment:
I don't quite understand this solution.
`userError` throws [IOError](https://hackage.haskell.org/package/base/docs/Prelude.html#v:userError) - how exactly does it help us not to lose the seed?

Reply:
It doesn't throw! It _returns_ an error where previously we'd throw one. I did check that it didn't work before abd works now.

Reply:
Gotcha! Forgot that `userError` simply constructs `IOError`, doesn't throw it.